File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/UpdateBlockEntityPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record UpdateBlockEntityPacket(int x, short y, int z, short action, byte[] nbtData) implements Packet.OUTGOING {

    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(x);
        os.writeShort(y);
        os.writeInt(z);
        os.writeByte(action);
        os.writeShort(nbtData.length);
        os.writeBytes(nbtData);
    }

    @Override
    public byte getId() {
        return 0x35;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/UpdateBlockEntityPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class UpdateBlockEntityPacket(val x: Int, val y: Short, val z: Int, val action: Short, val nbtData: ByteArray) :
    Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(x)
        os.writeShort(y.toInt())
        os.writeInt(z)
        os.writeByte(action.toInt())
        os.writeShort(nbtData.size)
        os.writeBytes(nbtData)
    }

    override fun getId(): Byte {
        return 0x35
    }
}
